utils/terrascript_client.py

```diff
@@ -435,17 +436,126 @@ def populate_tf_resource_s3(self, resource, namespace_info):
         # https://www.terraform.io/docs/providers/aws/r/s3_bucket.html
         values = {}
         values['bucket'] = identifier
-        values['versioning'] = {'enabled': True}
+        values['versioning'] = common_values.get('versioning') or False
         values['tags'] = common_values['tags']
         values['acl'] = common_values.get('acl') or 'private'
         values['server_side_encryption_configuration'] = \
             common_values.get('server_side_encryption_configuration')
         if common_values.get('lifecycle_rules'):
             # common_values['lifecycle_rules'] is a list of lifecycle_rules
             values['lifecycle_rule'] = common_values['lifecycle_rules']
+        sc = common_values.get('storage_class')
+        if sc:
+            rule = {
+                "id": sc + "_storage_class",
+                "enabled": "true",
+                "transition": {
+                    "storage_class": sc
+                }
+            }
+            if values.get('lifecycle_rule'):
```

Comment:
`common_values` already contains `lifecycle_rules`. is that a duplication?

Reply:
and can there be a lifecycle_rule without a storage_class?

Reply:
I have simplified the means to define a storage class for an s3 bucket done to just a single entry, but in fact it requires a whole lifecycle policy most of which the user doesn't need to know about.  This logic is handling appending the lifecycle rule for storage class if there are or are not already other lifecycle policies defined.

This could probably be removed if we push all the lifecycle definition to the config file.  That would have the benefit of making this simpler and support the few options this implementation isn't supporting, but shifts the knowledge of what is needed to enable a storage class to the user defining the config.

I'm not sure which one I prefer.